recipe/build.sh

```diff
@@ -11,49 +15,51 @@ if [[ $(uname) == "Darwin" ]]; then
 else
   export SONAME="-Wl,-soname,"
 fi
+
 # VERSION used in dylib versions in debian makefile patches
 export VERSION=$PKG_VERSION
 
-if [ "$PKG_NAME" == "scotch" ]
-then
-
-# build
-cd src/
-make esmumps 2>&1 | tee make.log
-cd ..
-# install
-mkdir -p $PREFIX/lib/
-cp -v lib/*${SHLIB_EXT}* $PREFIX/lib/
-mkdir -p $PREFIX/bin/
-cp -v bin/* $PREFIX/bin/
-mkdir -p $PREFIX/include/
-# avoid conflicts with the real metis.h
-mkdir -p include/scotch
-mv include/metis.h include/scotch/
-cp -rv include/* $PREFIX/include/
-
-fi # scotch
-
-
-if [ "$PKG_NAME" == "ptscotch" ]
-then
-
-export CCP=mpicc
-export CCD=${CCP}
-
-# build
-cd src/
-make ptesmumps 2>&1 | tee make.log
-cd ..
-# install
-mkdir -p $PREFIX/lib/
-cp -v lib/libpt*${SHLIB_EXT}* $PREFIX/lib/
-mkdir -p $PREFIX/bin/
-cp -v bin/dg* $PREFIX/bin/
-mkdir -p $PREFIX/include/
-cp -v include/ptscotch*.h $PREFIX/include/
-# avoid conflicts with the real parmetis.h
-mkdir -p $PREFIX/include/scotch
-cp -v include/parmetis.h  $PREFIX/include/scotch/
+
+if [[ "$PKG_NAME" == "scotch" ]]; then
+
+  # build
+  cd src/
+  make esmumps
+  cd ..
+
+  # install
+  mkdir -p $PREFIX/lib/
+  cp -v lib/*${SHLIB_EXT}* $PREFIX/lib/
+  mkdir -p $PREFIX/bin/
+  cp -v bin/* $PREFIX/bin/
+  mkdir -p $PREFIX/include/
+  # avoid conflicts with the real metis.h
+  mkdir -p include/scotch
+  mv include/metis.h include/scotch/
+  cp -rv include/* $PREFIX/include/
+
+elif [[ "$PKG_NAME" == "ptscotch" ]]; then
+
+  export CCP=mpicc
+  # This is only used by open-mpi's mpicc
+  # ignored in other cases
+  export OMPI_CC=$CC
+  export OPAL_PREFIX=$PREFIX
+
+  # build
+  cd src/
+  make ptesmumps
```

Comment:
If you do, then you don't need the new patch
```suggestion
  make ptesmumps CCD=$CC_FOR_BUILD
```

Reply:
If I understood https://github.com/conda-forge/scotch-feedstock/pull/67#issuecomment-1061875549 correctly, setting CCD might not be enough?

Reply:
I may be wrong (I am definititely not a Make expert) but in the end I went for the patch as it seemed to me that in https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L64 there was some recursive make logic that was basically overwriting the `CCD` value.

Reply:
For future reference, this fails: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=470850&view=logs&jobId=7afb22dc-3239-5456-794e-5378c2396929

```
+ make esmumps CCD=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_build_env/bin/x86_64-apple-darwin13.4.0-clang
(cd libscotch ;      make VERSION=6 RELEASE=0 PATCHLEVEL=9 scotch && make install)
make[1]: Entering directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work/src/libscotch'
make 					\
				CC="arm64-apple-darwin20.0.0-clang"					\
				CCD="arm64-apple-darwin20.0.0-clang"					\
				scotch.h					\
				scotchf.h					\
				libscotch.a					\
				libscotcherr.a				\
				libscotcherrexit.a
make[2]: Entering directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work/src/libscotch'
arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work=/usr/local/src/conda/scotch-6.0.9 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -fPIC -c arch.c -o arch.o -DSCOTCH_VERSION_NUM=6 -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_PATCHLEVEL_NUM=9
arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work=/usr/local/src/conda/scotch-6.0.9 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -DSCOTCH_VERSION_NUM=6 -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_PATCHLEVEL_NUM=9 dummysizes.c -o dummysizes -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -L/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -lm -lz -pthread -lm -lz -pthread -lm -lz -pthread
./dummysizes "-s" library.h scotch.h
/bin/sh: ./dummysizes: Bad CPU type in executable
```

Reply:
Perhaps a simpler patch could be to remove the line https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L64, but to be honest I am not 100% sure of the other side effects of that patch, while the existing patch of using `CC_FOR_BUILD` only when I know it was necessary it was more clear to me.